opt-env-conf/src/OptEnvConf/Doc.hs

```diff
@@ -400,32 +414,32 @@ renderVersionPage progname version =
 -- | Render the output of @--help@
 renderHelpPage :: String -> String -> AnyDocs SetDoc -> [Chunk]
 renderHelpPage progname progDesc docs =
-  unlinesChunks
-    [ usageChunk : renderShortOptDocs progname (docsToOptDocs docs),
-      [],
-      unlinesChunks $ progDescLines progDesc,
-      headerChunks "Available settings",
-      renderSetDocs docs
-    ]
+  unlinesChunks $
+    concat
+      [ [ usageChunk : renderShortOptDocs progname (docsToOptDocs docs),
+          [],
+          unlinesChunks $ progDescLines progDesc,
+          headerChunks "Available settings",
+          renderSetDocs docs
+        ],
+        concat
+          [ [ headerChunks "Available commands",
+              renderCommandDocsShort docs
+            ]
+            | not (null (docsToCommandDocs docs))
+          ]
+      ]
 
 renderSetDocs :: AnyDocs SetDoc -> [Chunk]
 renderSetDocs = unlinesChunks . go
   where
     go :: AnyDocs SetDoc -> [[Chunk]]
     go = \case
-      AnyDocsCommands cs -> concatMap goCommand cs
+      AnyDocsCommands _ -> [] -- todo: this empty list causes the double newline between settings and commands
```

Comment:
This is a part that annoys me a bit, this is the line that causes the double newline between the settings and the commands block but so far I failed to find a way to avoid it without removing other empty lines

Reply:
@L7R7 This is a very common problem when producing output.
The "usual" way of dealing with it is to wrap the result in another layer of `Maybe` or `[]` so you can remove that layer before rendering.

Reply:
I tried that, but I failed. I was sure several times that I have the right solution but the tests disagreed :smile: